pkg/go/README.md

```diff
@@ -0,0 +1,83 @@
+# Federation Go Packages
+
+Core implementation packages for building Federation services in Go.
+
+## Package Structure
+
+```
+pkg/go/
+├── auth/          # Authentication and authorization
```

Comment:
I dropped auth from this particular commit - as I'm not yet using it in the example use I'll be committing in follow-up. I should consider dropping documentation discussion of auth... Or drop in Auth now.

Reply:
I'm going to leave Auth docs in here for now - Auth coming soon.